packages/components/src/components/hds/text/types.ts

```diff
@@ -0,0 +1,58 @@
+export type HdsTextTags = Extract<keyof HTMLElementTagNameMap, 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'div'>;
+
+export enum HdsTextColorValues {
+  Primary = 'primary',
+  Strong = 'strong',
+  Faint = 'faint',
+  Disabled = 'disabled',
+  HighContrast = 'high-contrast',
+  Action = 'action',
+  ActionHover = 'action-hover',
+  ActionActive = 'action-active',
+  Highlight = 'highlight',
+  HighlightOnSurface = 'highlight-on-surface',
+  HighlightHighContrast = 'highlight-high-contrast',
+  Success = 'success',
+  SuccessOnSurface = 'success-on-surface',
+  SuccessHighContrast = 'success-high-contrast',
+  Warning = 'warning',
+  WarningOnSurface = 'warning-on-surface',
+  WarningHighContrast = 'warning-high-contrast',
+  Critical = 'critical',
+  CriticalOnSurface = 'critical-on-surface',
+  CriticalHighContrast = 'critical-high-contrast',
+}
+export type HdsTextColors = `${HdsTextColorValues}`;
+
+export enum HdsTextAlignValues {
+  Left = 'left',
+  Center = 'center',
+  Right = 'right',
+}
+export type HdsTextAligns = `${HdsTextAlignValues}`;
+
+export enum HdsTextWeightValues {
+  Regular = 'regular',
+  Medium = 'medium',
+  Semibold = 'semibold',
+  Bold = 'bold',
+}
+export type HdsTextWeights = `${HdsTextWeightValues}`;
+
+export enum HdsTextSizeValues {
+  OneHundred = 100,
+  TwoHundred = 200,
+  ThreeHundred = 300,
+  FourHundred = 400,
+  FiveHundred = 500,
+}
+type HdsTextSizesString = `${HdsTextSizeValues}`;
+type HdsTextSizesNumber = `${HdsTextSizeValues}` extends `${infer T extends number}` ? T : never;
+export type HdsTextSizes = HdsTextSizesString | HdsTextSizesNumber;
```

Comment:
I'm on board to move it to just strings! Just want to flag that this might be a breaking change as the component currently does accept number args? https://github.com/hashicorp/design-system/blob/65945c48775f3afaf98a254c7ba578484f78f72b/packages/components/src/components/hds/text/body.ts#L54-L57

Reply:
🤦 you're right, we do seem to be supporting both (albeit not being documented); your approach makes perfect sense then – sorry for not checking this beforehand. let me know if you'd like me to have a look at failing test

Reply:
Np at all! Thanks for the offer re: the failing test, I _think_ it's 🟢 now!